src/main/java/nextstep/subway/domain/Section.java

```diff
@@ -40,4 +53,114 @@ public void belongLine(Line line) {
     public Line getLine() {
         return this.line;
     }
+
+    public String findUpStationName() {
+        return upStation.getName();
+    }
+
+    public String findDownStationName() {
+        return downStation.getName();
+    }
+
+    public boolean isSameUpStationId(Section section) {
+        return upStation.getId() == section.findUpStationId();
+    }
+
+    public boolean isSameDownStationId(Section section) {
+        return downStation.getId() == section.findDownStationId();
+    }
+
+    public boolean isSameDownStationId(Station station) {
+        return this.downStation.getId() == station.getId();
+    }
+
+    public boolean isSameUpStationId(Station station) {
+        return this.upStation.getId() == station.getId();
+    }
+
+    public void updateAndCreateTwiceSectionWhenUpStationSame(Section newSection) {
+        checkDistanceValidation(this.distance, newSection.distance);
+        this.modifyUpStation(newSection.downStation);
+
+        this.modifyDistance(this.distance, newSection.distance);
+    }
+
+    public void updateAndCreateTwiceSectionWhenDownStationSame(Section newSection) {
+        checkDistanceValidation(this.distance, newSection.distance);
+        this.modifyDownStation(newSection.upStation);
+
+        newSection.modifyDistance(this.distance, newSection.distance);
+    }
+
+    private void modifyDownStation(Station tempStation) {
+        this.downStation = tempStation;
+    }
+
+    private void modifyUpStation(Station downStation) {
+        this.upStation = downStation;
+    }
+
+    private void modifyDistance(long existingDistance, Long distance) {
+        this.distance = existingDistance - distance;
+    }
+
+    private void checkDistanceValidation(Long existingDistance, Long newDistance) {
+        if (existingDistance < newDistance || existingDistance == newDistance) {
```

Comment:
```suggestion
        if (existingDistance <= newDistance) {
```
이렇게 작성해줄 수 있지 않을까요..?